### PR TITLE
fix(material/paginator): allow readonly options (#24050)

### LINKED ISSUE #24050: bug(MatPaginator): pageSizeOptions are required to be mutable Array, ReadonlyArray fails
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

_No response_

### Description

MatPaginator requires `pageSizeOptions` to be a mutable array. Passing in a ReadonlyArray<number> fails.

### Reproduction

Steps to reproduce:
```typescript
@Component({
  selector: 'example-component',
  template: '<mat-paginator [pageSizeOptions]="pageSizeOptions"></mat-paginator>',
})
export class ExampleComponent {
 pageSizeOptions: ReadonlyArray<number> = [10, 20];
}
```
https://stackblitz.com/edit/components-issue-rvmtli?file=src/app/example-component.ts

### Expected Behavior

I am unaware that MatPaginator would mutate the passed in Array. Passing in a ReadonlyArray<number> should work.

### Actual Behavior

Throws

```
The type 'readonly number[]' is 'readonly' and cannot be assigned to the mutable type 'number[]'.
```

### Environment

- Angular: ^12.2.9
- CDK/Material: ^12.2.9
- TypeScript: ^3.7.5
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): macOS

### PR DESCRIPTION
**Pull request by lekhmanrus**
Adds `readonly` support for paginator `pageSizeOptions`.

closes #24050

### CODE REVIEW

File: src/material/paginator/paginator.ts [hunk 1]
```diff
@@ -146,14 +146,14 @@ export abstract class _MatPaginatorBase<
 
   /** The set of provided page size options to display to the user. */
   @Input()
-  get pageSizeOptions(): number[] {
+  get pageSizeOptions(): readonly number[] {
```
[the remark this answers is not in the document]
Comment (lekhmanrus) [reply src/material/paginator/paginator.ts]: Right, fixed. Thanks @crisbeto 🙂

[the remark this answers is not in the document]
Comment (crisbeto) [reply src/material/paginator/paginator.ts]: I believe that this is still a breaking change for the example I mentioned above. See https://www.typescriptlang.org/play?#code/DYUwLgBAZg9jBcEB2BXAtgIxAJwNoF0IAfCbEAQwBMYlgBPZdLPfAbgCh3YYIBeCAh24A6AA4oAzgAsAFAEYAlKyA

[the remark this answers is not in the document]
Comment (lekhmanrus) [reply src/material/paginator/paginator.ts]: we have `number[] | readonly number[]` for setter only, and still have just `number[]` for getter. So, it should work as expected. See [TS Playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAECC0G8BQ1oHMCmAXaAHMmAygJYBeGA8rlsQPYB2EAFAJQBc09ArgLYBGGAE4BtALqIUqaIOxdB9aFgAWxCADoA+viJlK1OowDckgL6SI2PAQwlyVGg2YA3MCC4YO3fkLHQAPtIYYAAmDCAAnpy8AiKiLBJSiirqWta2eg6M0AC80MJqBS5uGKKmkriCxC5YGNCpOnb6jp7RPuK5YsZmSMCOOABmtLQ5nBgA7nCsxoO0ato2uvYG6rhcEEpMAIwshkA).

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**